src/actions/commands/actions.ts

```diff
@@ -3393,40 +3393,46 @@ class ActionJoinNoWhitespace extends BaseCommand {
   modes = [Mode.Normal];
   keys = ['g', 'J'];
   canBeRepeatedWithDot = true;
-  runsOnceForEachCountPrefix = true;
 
   // gJ is essentially J without the edge cases. ;-)
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    if (position.line === TextEditor.getLineCount() - 1) {
+    const line = position.line;
+    if (line === TextEditor.getLineCount() - 1) {
       return vimState; // TODO: bell
     }
 
-    let lineOne = TextEditor.getLineAt(position).text;
-    let lineTwo = TextEditor.getLineAt(position.getNextLineBegin()).text;
-
-    lineTwo = lineTwo.substring(
-      TextEditor.getFirstNonWhitespaceCharOnLine(position.getDown().line).character
-    );
-
-    let resultLine = lineOne + lineTwo;
+    const count = vimState.recordedState.count > 2 ? vimState.recordedState.count - 1 : 1;
+    const lastLine = Math.min(line + count, TextEditor.getLineCount() - 1);
+    const trimmedLines: string[] = [];
+    for (let i = line + 1; i <= lastLine; i++) {
+      const text = TextEditor.getLine(i).text;
+      trimmedLines.push(text.substring(TextEditor.getFirstNonWhitespaceCharOnLine(i).character));
```

Comment:
I think we had this wrong before; `gJ` doesn't seem to remove any whitespace:

```
gJ			Join [count] lines, with a minimum of two lines.
			Don't insert or remove any spaces.
```

Reply:
You're completely right. I'll try to fix it.

Reply:
I fixed it.